### ISSUE TITLE
'Open' directory toggle

### ISSUE THREAD

**Issue by boonebgorges** [cuny-academic-commons/commons-in-a-box#227]
Related to #211. See http://redmine.citytech.cuny.edu/issues/2738

The OpenLab will be adding an 'Open' toggle to the directory filters. The toggle will filter results to those groups that are `public`, and if they have sites, they must have `blog_public=1`. To make it feasible to filter, this probably means mirroring a site's `blog_public` value as groupmeta on the corresponding group.

Before this can be implemented on CBOX, we need to have a migration routine that will allow us to backfill the `blog_public` data in this way. As such, we can only add this for 1.2.0 if we first do #216.

**Comment by boonebgorges** [comment]
@Mamaduka I wanted to let you know that the task in 0d98d7b took me about 3 minutes to write because the upgrades system is so easy to use :-D

**Comment by Mamaduka** [comment]
@boonebgorges amazing. Looking forward to more internal use cases and see how we can improve the API. But I think we have a solid start.

**Comment by bree-z** [comment]
Hi Boone and George,

I just wanted to double check that this is supposed to work differently than on the OpenLab. From http://redmine.citytech.cuny.edu/issues/2738 

> Note that all of the sidebar filters use "intersect" logic, not "union". That is, when you add another filter, you are drilling down your results, not expanding them. ("and" not "or") This is pretty intuitive in most cases, though in the case of badges it's not quite as clear. If you check more than one badge box, you will only get results that have all of the checked badges.
  
Is the logic a "union" logic here, rather than "intersect"? For example, if I search for courses with the keyword "politics" and badges 'essc' and 'wac', I get all courses with the keyword "politics" that have either 'ESSC' or 'WAC' badges. As in the first screenshot.

But, something is also off with the 'Open' badge. If I search for Courses with an 'Open' badge, I get at least a few that are not open, e.g. https://openlabdev.commonsinabox.org/groups/cat-curling-1101/ and https://openlabdev.commonsinabox.org/groups/bio-101/

Thanks!

### ESSC / WAC search:

![Search_PoliticsESSCWAC](https://user-images.githubusercontent.com/7189824/89739438-b6f47600-da4e-11ea-9135-9a7f0aef4ef1.png)

### Open Search:

![Search_CoursesOpen](https://user-images.githubusercontent.com/7189824/89739450-c8d61900-da4e-11ea-96e1-aca2d8745df0.png)

**Comment by boonebgorges** [comment]
First, it looks like part of the 'Open' filter logic was not properly ported from the OpenLab. 'Open' means that the group is public AND the site is public; but I neglected to merge the part that looked at the site. This should now be fixed, which should take care of "something is off with the 'Open' badge".

As for the broader logic: The relationship of the sidebar filters to one another is, indeed, "intersect" and not "union". But the relationships *within* individual filters is "union". This applies specifically to the checkbox filters of 'Badges' and 'Group Type', each of which operate as a single filter. Checking more than one Badge will return groups matching *any* of the checked badges; mutatis mutandis for Group Types. But if you combine a Badge with a School, you'll get only results that match *both*. (And if you combine *two* Badges with a School, you'll get any item that matches both the School and one or both of the Badges.)

**Comment by bree-z** [comment]
Thanks, Boone!

I still see the same issue with 'Open' badges.  When I filter for 'Open' courses, I still get the same results as shown in the screenshot in the comment above, including some with private sites.

**Comment by boonebgorges** [comment]
It's possible the latest changes hadn't been pulled to openlabdev.commonsinabox.org. Please try again https://openlabdev.commonsinabox.org/groups/type/course/?search=&academic-unit-schools=&term=&sort=alphabetical&is_open=1#

**Comment by bree-z** [comment]
That was it, thanks! The 'Open' filter is now being applied properly.

The more general logic also looks good.

**Comment by bree-z** [comment]
Sorry, Boone. I found another strange thing with the badge.  Existing groups that have public groups and sites with the setting 'Ask search engines not to index this site. The site should not show up in web search results.' have the 'Open' badge applied.  E.g.: https://openlabdev.commonsinabox.org/groups/language-culture-and-society/

However, when I cloned the group above without changing the site privacy settings, the 'Open' badge was not applied. I changed the site settings for this group to 'Allow search engines to index this site. The site will show up in web search results. Then, the 'Open' badge appeared. I changed the settings *back* to 'Ask search engines not to index this site. The site should not show up in web search results.' and the badge remained. Here's the URL: https://openlabdev.commonsinabox.org/groups/language-culture-and-society-clone/ 

Then, I cloned the site above once more and kept the site settings of 'Ask search engines not to index this site. The site should not show up in web search results.' The badge doesn't appear. But I didn't change anything. HEre's the URL: https://openlabdev.commonsinabox.org/groups/clone-of-language-culture-and-society-clone/

Thanks!
